modules/internal/ChapelLocale.chpl

```diff
@@ -247,6 +247,35 @@ module ChapelLocale {
       return "";
     }
 
+    //
+    // Support for different types of memory:
+    // large, random-access (low latency), and streaming (high bandwidth)
+    // Inspired by madvise()'s support for RANDOM and SEQUENTIAL access
+    //
+    pragma "no doc"
+    proc defaultMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc largeMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc randomMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc streamingMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
```

Comment:
Adding on to what Greg is saying, I thought we were trying to make getChildCount / getChild take in an optional bitmask argument indicating the type of resources requested. I think we were hoping that large/random/streaming memory would be bitmask positions and that we'd have functions to gather up sublocales with a particular combination of properties.

I don't think what you have here is necessarily the wrong direction; I just think we'd like to possibly have a different user interface that's got more of a flavor of querying for the sublocale with the best match for some pattern of expected use.